File: src/org/softeg/slartus/forpdaplus/controls/KeyEventEditPreference.java
package org.softeg.slartus.forpdaplus.controls;

import android.content.Context;
import android.preference.EditTextPreference;
import android.util.AttributeSet;
import android.view.KeyEvent;
import android.view.View;

/**
 * Created by slinkin on 14.01.14.
 */
public class KeyEventEditPreference extends EditTextPreference {
    public KeyEventEditPreference(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);

    }

    public KeyEventEditPreference(Context context, AttributeSet attrs) {
        super(context, attrs);

    }

    public KeyEventEditPreference(Context context) {
        super(context);

    }

    @Override
    protected void showDialog(android.os.Bundle state) {
        super.showDialog(state);

        getEditText().setOnKeyListener(new View.OnKeyListener() {
            @Override
            public boolean onKey(View view, int i, KeyEvent keyEvent) {
                getEditText().setText(Integer.toString(keyEvent.getKeyCode()));
                return false;
            }
        });
    }

}

File: feature/feature-preferences/src/main/java/org/softeg/slartus/forpdaplus/feature_preferences/views/KeyEventEditPreference.kt
package org.softeg.slartus.forpdaplus.feature_preferences.views

import android.content.Context
import android.util.AttributeSet
import android.widget.EditText
import androidx.preference.EditTextPreference
import org.softeg.slartus.forpdaplus.feature_preferences.R

import org.softeg.slartus.forpdaplus.feature_preferences.getAttr

class KeyEventEditPreference @JvmOverloads constructor(
    context: Context?,
    attrs: AttributeSet?,
    defStyleAttr: Int = context.getAttr(
        R.attr.editTextPreferenceStyle,
        android.R.attr.editTextPreferenceStyle
    ),
    defStyleRes: Int = 0
) : EditTextPreference(
    context,
    attrs,
    defStyleAttr,
    defStyleRes
) {
    fun init() {
        setOnBindEditTextListener { editText: EditText? ->
            editText?.setOnKeyListener { _, _, keyEvent ->
                editText.setText(keyEvent?.keyCode?.toString())
                false
            };
        }
    }
}